src/app/api/metadata/route.ts

```
@@ -0,0 +1,179 @@
+      !url.hostname.match(/^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/)
+    ); // Block IP addresses
+  } catch {
+    return false;
+  }
+}
+
+// Sanitize metadata fields
+function sanitizeMetadata(metadata: Record<string, string | undefined>): Record<string, string> {
+  return Object.entries(metadata).reduce(
+    (acc, [key, value]) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)